connector/connect/common/src/main/protobuf/spark/connect/base.proto

```diff
@@ -271,15 +271,31 @@ message ExecutePlanRequest {
   // collate streaming responses from different queries within the dedicated session.
   string session_id = 1;
 
+  // (Optional)
+  // Provide an id for this request. It is needed if the request should be retryable.
+  optional string operation_id = 5;
+
   // (Required) User context
   //
   // user_context.user_id and session+id both identify a unique remote spark session on the
   // server side.
   UserContext user_context = 2;
 
-  // (Required) The logical plan to be executed / analyzed.
+  // The logical plan to be executed / analyzed.
+  // If the query has already been submitted, and this is a retry request, the plan is ignored.
   Plan plan = 3;
 
+  // (Optional)
+  // Specify if the request should be reattachable after e.g. a broken connection.
+  // If this is set to true, the client should call ReleaseRequest after it is done to let the
+  // server release resources; otherwise resources would be released only after a timeout.
+  optional bool reattachable = 6;
+
+  // (Optional)
+  // If specified, start streaming the query results from a given row offset.
+  // (Useful when this is a reattach request).
+  optional int64 start_offset = 7;
```

Comment:
Semantically a bit confusing if this is not a reattach. e.g., what does it mean if I send `Execute("select * from tbl", start_offset=100)` on my initial request. Am I saying that I only want to receive rows > 100 on this execution?

Reply:
Yep, would be cleaner if continuation is a separate RPC.
Maybe a bi-directional streaming, where the client could better control the flow of data:
- the client could send a request to just stream all results
- the client could send a request to send at most X rows/bytes from given offset
- the client could send a request saying that results up to offset X have been received and no longer need to be retained by the server anymore.

Reply:
What we also discussed with @grundprinzip is that many of the ExecutePlanRequests are Commands which are not a stream of Arrow results, but that are Commands, which return an arbitrary stream of ExecutePlanResponse.
So these requests to get results may not be based on offsets in the Arrow results, which is specific for queries that return arrow results, but rather on the ExecutePlanResponse: "transmit me the stream of ExecutePlanResponse starting from Nth response".
It's also possible to have different kind of requests on the request stream: both a generic one ("transmit the stream from nth ExecutePlanResponse") and more specific one ("send a response with the schema"; "send arrow batches from offset N").
For translating a thrift FetchResults request you would need these latter specialized reqs.

We also discussed yesterday with Martin, that for the "get me results from offset N", we would not want to break / reserialize Arrow batches. So, if the offset is not a starting offset of an Arrow batch, it will send an arrow batch from whenever that arrow batch starts, and send what the actual offset was. Luckily, this is what FetchResults in thrift already does.